### DISCUSSION TITLE
Bedrock: Ask to support Claude Computer Use in Converse Function

### DISCUSSION THREAD

**Discussion by skerava** [aws/aws-sdk-go-v2#2898] [General]
### Describe the feature

* It is very normal and convenient to use 'converse' method to invoke models in Bedrock.Many system integrations are based on the 'converse' method.
* Currently, the AWS Bedrock API and Python SDK support the 'converse' method to invoke the 'computer use' feature of the Claude Haiku 3.5/Sonnet 3.5 v2 model.
https://aws.amazon.com/cn/blogs/aws/upgraded-claude-3-5-sonnet-from-anthropic-available-now-computer-use-public-beta-and-claude-3-5-haiku-coming-soon-in-amazon-bedrock/
* However it seems that go sdk now only support ‘computer use’ feature in InvokeModel Method.

Please integrate support in the Go SDK to use the 'converse' method to invoke Claude's 'computer use' feature.

### Use Case

Same above

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### AWS Go SDK V2 Module Versions Used

github.com/aws/aws-sdk-go-v2 v1.27.1

### Go version used

1.23.3

**Comment by RanVaknin** [comment]
Hi @skerava,

Could you clarify what specific SDK implementation you're looking for? The computer use feature should already work with the existing `InvokeModel` method by passing the appropriate JSON request body (including the `tools` configuration and `anthropic_beta` flag with the new computer use value as shown in the blogpost). 

Thanks,
Ran~

**Comment by skerava** [comment]
Hi @RanVaknin,

Here are the key points:

The Bedrock runtime supports two methods for interacting with Claude models:**Converse** and **InvokeModel**. Our system, as well as those of our customers, are both based on the **Converse** method.

As you mentioned, the computer-use feature currently works with the InvokeModel method. However, it is **not yet supported** with the Converse method, even though **AWS API** and **Python's boto3** library already support it.

Our customer's system has been waiting for the Bedrock-runtime's Converse method in the Go SDK to support the computer-use feature, as they already have **clear project goals** that require this capability. Reintegrating with the InvokeModel method would not only increase the system's complexity but also fall short of customer expectations, as they do not want to integrate the same functionality twice.

**Reply by RanVaknin** [reply]
Hi @skerava ,

The point I was trying to make is that since this flag is supported server side it should work on any SDK. Since you send that flag as a raw json string body of the request, the SDK cannot "add" this capability, its either supported server side or it isn't.

The converse [API docs](https://docs.aws.amazon.com/bedrock/latest/userguide/computer-use.html) show that this is indeed supported but the request structure is different between `Converse` and `InvokeModel`. Are you structuring your request correctly?

>Python's boto3 library already support it.

Can you please share a code snippet where you are calling this method with the boto3 SDK (with request and response logs enabled), and then the exact same request with the Go SDK (with request and response logs enabled as well).

Thanks,
Ran~
